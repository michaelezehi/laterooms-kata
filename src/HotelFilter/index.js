import React, { Component } from 'react';
import { array, func } from 'prop-types';

import { FormGroup, Checkbox } from 'react-bootstrap';

class HotelFilter extends Component {

  static propTypes = {
    filters: array.isRequired,
    updateFilters: func.isRequired,
    filterResults: func.isRequired,
  };

  addFilter(value) {
    const filtersArr = this.props.filters;

    const valueIndex = filtersArr.indexOf(value);

    // Only add filter if not already present
    if (valueIndex === -1) {
      filtersArr.push(value);
    }

    return filtersArr;
  }

  removeFilter(value) {
    const filtersArr = this.props.filters;

    const valueIndex = filtersArr.indexOf(value);

    if (valueIndex !== -1) {
      filtersArr.splice(valueIndex, 1);
    }

    return filtersArr;
  }

  handleFilterChange() {
    if (this.carParkActive.checked) {
      this.props.updateFilters(this.addFilter('car park'));
    } else {
      this.props.updateFilters(this.removeFilter('car park'));
    }

    if (this.poolActive.checked) {
      this.props.updateFilters(this.addFilter('pool'));
    } else {
      this.props.updateFilters(this.removeFilter('pool'));
    }

    if (this.gymActive.checked) {
      this.props.updateFilters(this.addFilter('gym'));
    } else {
      this.props.updateFilters(this.removeFilter('gym'));
    }

    this.props.filterResults();
  }

  render() {
    return (
      <FormGroup>
        <Checkbox onChange={this.handleFilterChange.bind(this)} inputRef={ref => { this.carParkActive = ref; }} inline>Car Park</Checkbox>
        <Checkbox onChange={this.handleFilterChange.bind(this)} inputRef={ref => { this.poolActive = ref; }} inline>Pool</Checkbox>
        <Checkbox onChange={this.handleFilterChange.bind(this)} inputRef={ref => { this.gymActive = ref; }} inline>Gym</Checkbox>
      </FormGroup>
    )
  }
}

export default HotelFilter;
