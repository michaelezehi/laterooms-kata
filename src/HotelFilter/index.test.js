import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import HotelFilter from './index';

import { HOTELS } from '../hotelData';

// TODO - move to mocks
const FILTERS = [];
const HOTELLIST = HOTELS;

const updateFilters = (val) => {
  FILTERS = val;
};

it('renders without crashing', () => {
  const component = renderer.create(
    <HotelFilter
      filters={FILTERS}
      updateFilters={updateFilters}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('adds filter to filters arr', () => {
  const component = shallow(
    <HotelFilter
      filters={FILTERS}
      updateFilters={updateFilters}
    />,
  );

  component.instance().addFilter('car park');
  expect(FILTERS[0]).toBe('car park');
  expect(FILTERS.length).toBe(1);
});

it('adds removes filter from filters arr', () => {
  const component = shallow(
    <HotelFilter
      filters={FILTERS}
      updateFilters={updateFilters}
    />,
  );

  expect(FILTERS[0]).toBe('car park');
  expect(FILTERS.length).toBe(1);

  component.instance().removeFilter('car park');
  expect(FILTERS.length).toBe(0);
});
