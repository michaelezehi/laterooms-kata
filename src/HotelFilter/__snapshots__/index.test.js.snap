// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor={undefined}
  >
    Filter:
  </label>
  <br />
  <label
    className="checkbox-inline"
    style={undefined}
    title=""
  >
    <input
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    Car Park
  </label>
  <label
    className="checkbox-inline"
    style={undefined}
    title=""
  >
    <input
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    Pool
  </label>
  <label
    className="checkbox-inline"
    style={undefined}
    title=""
  >
    <input
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    Gym
  </label>
</div>
`;
